controllers/movies.js

```diff
@@ -0,0 +1,41 @@
+const movies = require('../movies')
+
+const getAllMovies = (req, res) => {
+  res.send(movies)
+}
+
+const searchByTitleAndDirector = (req, res) => {
+  const qSearch = req.params.query.toLowerCase()
+
+  const movieResults = movies.filter(movie => movie.title.toLowerCase().includes(qSearch))
+
+  const directorResults = movies.filter(movie => {
+    for (director of movie.directors) {
+      if (director.toLowerCase().includes(qSearch)) {
+        return movie
```

Comment:
Better to return true or false instead of movie object or nothing.

Although filter fn will understand/translate movie as true and nothing as false, it's not clear at first sight

Reply:
also why return in middle of the loop... what if multiple movies have same text in the title

Lord of the rings 1
Lord of the rings 2
Lord of the rings 3

and if you search lord... it should return all 3.